sdk/core/src/metadata.rs

```diff
@@ -0,0 +1,47 @@
+use crate::AddAsHeader;
+use http::request::Builder;
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+#[derive(Debug, Clone)]
+pub struct Metadata<'a>(HashMap<Cow<'a, str>, Cow<'a, str>>);
```

Comment:
Same point here about `Cow`. This makes the type harder to understand and thus harder to use.

Reply:
I am on the fence on whether to accept `&str` or a `String`. `&str` means the `Metadata` struct cannot escape the owning function easily while forcing `string` means potentially copying hundreds of strings around. 

I'll go with `String` for ergonomics.

Reply:
For everything else we're taking &str. I'm fine with going with `String`. We might want to think about why `Metadata` is special and takes `String`. Perhaps more types should take `String` instead of `&str`. Something to think about....

Reply:
My reasoning is: for something that does not make sense outside a request (like `client_request_id`) `&str` is fine. This is because you use it on a request only, and a requests are short lived 99% of the time (so borrowing makes sense).

For everything else `String` is better.